packages/astro/src/content/types-generator.ts

```diff
@@ -362,6 +362,28 @@ function invalidateVirtualMod(viteServer: ViteDevServer) {
 	viteServer.moduleGraph.invalidateModule(virtualMod);
 }
 
+/**
+ * Takes a configPath and returns a normalized relative version:
+ *  -   If is not relative, it adds `./` to the beginning.
+ *  -   If it ends with `.ts`, it removes the extension.
+ *  -   It adds `.js` to the end.
+ *  -   It stringifies the result (adds `""` around it).
+ * @param from Config path from path.
+ * @param to Config path to path.
+ * @returns Normalized config path.
+ */
+function normalizeConfigPath(from: string, to: string) {
+	const normalizedConfigPath = path.relative(from, to);
+
+	return JSON.stringify(
+		`${isRelativePath(normalizedConfigPath) ? '' : './'}${
+			normalizedConfigPath.endsWith('.ts')
```

Comment:
There are also `.d.ts` files. Are they covered by this check on purpose?

Reply:
AFAIK it should not be .d.ts? So is it worth handling this case?

Reply:
If we never get `.d.ts` and is always `.ts` then we can update this to replace `.ts` with `.js` simplifying the code a bit:

```ts
function normalizeConfigPath(from: string, to: string) {
	const normalizedConfigPath = path.relative(from, to);

	return JSON.stringify(
		`${isRelativePath(normalizedConfigPath) ? '' : './'}${
			normalizedConfigPath.replace(/\.ts$/, '.js')
		}`
	);
}
```

Reply:
> If we never get .d.ts

That's what I **not** sure about (I am not familiar with this part of the code). So maybe keeping the code as is better.

Reply:
cc @bholmesdev

Reply:
Ah, this looks like the loader for `src/content/config.ts` correct? If so, we only support `.ts` and `.js`. `.d.ts` shouldn't be possible 👍